tests/rptest/tests/partition_movement_test.py

```diff
@@ -368,7 +368,19 @@ def test_bootstrapping_after_move(self, num_to_upgrade):
 
         # snapshot offsets
         rpk = RpkTool(self.redpanda)
-        partitions = rpk.describe_topic(spec.name)
+
+        def has_offsets_for_all_partitions(out):
+            # NOTE: partitions may not be returned if their fields can't be
+            # populated, e.g. during leadership changes.
+            partitions = list(rpk.describe_topic(spec.name))
```

Comment:
nit: something to keep in mind: i don't think that users of rpk.describe_topic shouldn't have to deal with this. a better solution would be to either make describe_topics be more robust, or signal to the caller that the results may be incomplete, or some other reasonable protocol.

Reply:
Yeah, overall I agree, gonna punt on this for now because I don't want it to block these tests